libs/rtefsutils/src/RteFsUtils.cpp

```diff
@@ -25,6 +25,7 @@
 #include <iomanip>
 #include <regex>
 #include <thread>
+#include <sys/stat.h>
```

Comment:
This is not platform independent and fails at least on Windows. We can't use `umask`, at least not directly.

Reply:
Well, okay. Not sure about that. The Windows builds are broken. Perhaps we need to add additional library to the project?

Reply:
Appears that the Windows umask function does not behave like a normal UNIX umask, so maybe it's better to skip the umask fallback on Windows?

Reply:
Well, we need to abstract this functionality if we can't rely on the c/c++ libraries. As a coding guideline we have defined to not use platform dependent code (via `#ifdef`'s) anywhere than in the CrossPlatform lib. We'd need to add a `umask` function to `CrossPlatformUtils` and implement it for all platforms. For Windows we might use some static defaults if we can't retrieve the setting.

Reply:
I created a function GetCurrentUmask() that returns the current umask as a std::filesystem::perms instance. For Windows, I mapped the bits to the owner, but perhaps it should be mapped to (owner | group | other).

Please take a look and see if this is better. :)

Reply:
What's you view on it @JonatanAntoni, is it okay like it is now?

Reply:
I think its okay, even if one could consider the solution over-engineered.

Perhaps, instead of using `umask` we could simply fall back to owner-only as this is what we need to make changes. I think using owner-only on Windows is fine.

Reply:
So, do you want me to change it to be hard-coded `0700` on Windows and leave the platforms at `0777 & ~umask()` or should all of them be hard-coded or keep it as it is now?
I would really want to get this merged as it's supper annoying to run the test-suite locally when files are no longer accessible...

Reply:
Lets keep it as is for now and wait for @edriouk.

I appreciate the annoyance of this but in light of the planned 2.0.0 release I'd rather like to be super conservative. I feel the same on Mac here, because I need to run git clean through sudo.

If this is not an immediate bug blocking features in 2.0.0, I'd give us some more days to conclude properly.

Reply:
Ok. Let's give @edriouk a few days to review and revisit the topic early next week.

I think the security implications with files "randomly" becoming world writable (#1006) is something that should be fixed before the 2.0.0 as it could be dangerous for customer projects.

Reply:
I agree about the security implication. Unfortunately, this fix doesn't address all the issues but only parts of it. And since pack installation is now handled by `cpackget` the fix in RteUtils don't have any impact.

We need to clarify in what situations these functions are effectively used in released tools.